services/consensusalgo/benchmarkconsensus/common.go

```diff
@@ -30,18 +30,20 @@ func (s *Service) getLastCommittedBlock() (primitives.BlockHeight, *protocol.Blo
 	return s.lastCommittedBlockUnderMutex.TransactionsBlock.Header.BlockHeight(), s.lastCommittedBlockUnderMutex
 }
 
-func (s *Service) setLastCommittedBlock(blockPair *protocol.BlockPairContainer, expectedLastCommittedBlockBefore *protocol.BlockPairContainer) error {
+func (s *Service) setLastCommittedBlockIfPreviousBlockMatches(blockPair *protocol.BlockPairContainer, expectedLastCommittedBlockBefore *protocol.BlockPairContainer) bool {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
 	if s.lastCommittedBlockUnderMutex != expectedLastCommittedBlockBefore {
-		return errors.New("aborting shared state update due to inconsistency")
+		s.logger.Info("setLastCommittedBlockIfPreviousBlockMatches ignoring block, expected previous block different than actual - probably updated concurrently from a different flow")
```

Comment:
we can return an error if the height of the expected last committed block is higher than the actual last committed block